assignment/final/aLife/code/master.js

```diff
@@ -0,0 +1,417 @@
+}
+
+function removeTBClicked() {
+    terminalButton.classList.remove("terminalButtonClicked");
+}
+
+function addTBClicked() {
+    terminalButton.classList.add("terminalButtonClicked");
+}
+
+function preBuiltPkmn(pkmn) {
+    //console.log("running in preBuiltPkmn()");
+    currentPkmn = pkmn;
+    selectNewPkmn(pkmn.NAME);
+    moveset.innerHTML = "";
+    terminalScreen.innerHTML =
+        `
+        <img src="${pkmn.SPRITES.front_default}">
+        `;
+    statSection.innerHTML =
+        `
+        <div class="col-sm-6">${pkmn.STATS[0].STAT}: ${pkmn.STATS[0].VALUE}</div>
+        <div class="col-sm-6">${pkmn.STATS[1].STAT}: ${pkmn.STATS[1].VALUE}</div>
+        <div class="col-sm-6">${pkmn.STATS[2].STAT}: ${pkmn.STATS[2].VALUE}</div>
+        <div class="col-sm-6">${pkmn.STATS[3].STAT}: ${pkmn.STATS[3].VALUE}</div>
+        <div class="col-sm-6">${pkmn.STATS[4].STAT}: ${pkmn.STATS[4].VALUE}</div>
+        <div class="col-sm-6">${pkmn.STATS[5].STAT}: ${pkmn.STATS[5].VALUE}</div>
+        `;
+
+    //console.log(pkmn.MOVES);
+    for (let i = 0; i < pkmn.MOVES.length; i++) {
+        moveset.innerHTML +=
+            `
+            <div class="row">
+                <div class="col-sm-5 offset-1">${pkmn.MOVES[i].NAME}</div>
+                <div class="col-sm-2">${pkmn.MOVES[i].TYPE}</div>
+                <div class="col-sm-1" style="text-align: right;">${pkmn.MOVES[i].POWER != null ? pkmn.MOVES[i].POWER : ""}</div>
+            </div>
+            `;
+    }
+
+    idBox.innerHTML = pkmn.ID;
+}
+
+function filterPkmn() {
+    const regex = new RegExp(".*" + pkmnName.value + ".*");
+    let count = pkmnList.filter(pkmn => pkmn.NAME.match(regex)).length;
+    pkmnListDiv.innerHTML = "";
+    for (let i = 0; i < count; i++) {
+        pkmnListDiv.innerHTML += `<p id="${pkmnList.filter(pkmn => pkmn.NAME.match(regex))[i].NAME}">${
+            pkmnList.filter(pkmn => pkmn.NAME.match(regex))[i].NAME}</p>`;
+    }
+}
+
+function filterByType() {
+    let alreadyAdded = [];
+
+    for(let i = 0; i < typesCorespondingPkmn.length; i++)
+    {
+        if(typesCorespondingPkmn[i].TYPE.TYPE == selectedType1.toLowerCase() ||
+        typesCorespondingPkmn[i].TYPE.TYPE == (selectedType2 != null? selectedType2.toLowerCase(): "")){
+            pkmnListForTypeFilter.push(typesCorespondingPkmn[i]);
+        }
+    }
+
+    if(selectedType2 == null){
+        //console.log(pkmnListForTypeFilter);
+        pkmnListDiv.innerHTML = "";
+        for(let i = 0; i < pkmnListForTypeFilter[0].LIST.length; i++){
+            if(alreadyAdded.indexOf(makingThisEasier(i)) < 0){
+                pkmnListDiv.innerHTML +=
+                `
+                <p id="${makingThisEasier(i) != ""? makingThisEasier(i): Math.random()}">
+                ${makingThisEasier(i) != ""? makingThisEasier(i): "Some pkmn from Alolan region are not included."}
+                </p>
+                `
+                alreadyAdded.push(makingThisEasier(i));
```

Comment:
Looks like `makingThisEasier` tries to get first part of a string up to character `-`...maybe...but you could cache the result off instead always going back to your function:
```
for(...){
 const lifeNowEasier = makingThisEasier(i);
// Now just use lifeNowEasier
}
```